CombinatorialGames/Nimber/Simplicity.lean

```diff
@@ -600,9 +605,126 @@ protected theorem IsAlgClosed.sSup {s : Set Nimber} (H : ∀ x ∈ s, IsAlgClose
     · rw [coeff_eq_zero_of_natDegree_lt hm]
       exact bot_le.trans_lt (hf n).2
 
+protected theorem IsNthDegreeClosed.iSup {n : ℕ} {ι} {f : ι → Nimber}
+    (H : ∀ i, IsNthDegreeClosed n (f i)) : IsNthDegreeClosed n (⨆ i, f i) := by
+  apply IsNthDegreeClosed.sSup
+  simpa
+
+/-- If `x` is a field, to prove it `n`-th degree closed, it suffices to check *monic* polynomials of
+degree less or equal to `n`. -/
+theorem IsNthDegreeClosed.ofMonic {n : ℕ} {x : Nimber} (h : IsField x)
+    (hp : ∀ p : Nimber[X], p.Monic → 0 < p.degree → p.degree ≤ n → (∀ k, p.coeff k < x) →
+      ∃ r < x, p.IsRoot r) : IsNthDegreeClosed n x where
+  has_root' p hp₀ hpn hp' := by
+    have hp₀' : p ≠ 0 := by rintro rfl; simp at hp₀
+    have hm : (C p.leadingCoeff⁻¹ * p).Monic := by simp [Monic, hp₀']
+    have hd : (C p.leadingCoeff⁻¹ * p).degree = p.degree := by compute_degree!
+    have := hp _ hm (hd ▸ hp₀) (hd ▸ hpn) fun k ↦ ?_
+    · aesop
+    · rw [coeff_C_mul, inv_mul_eq_div]
+      exact h.div_lt (hp' k) (hp' _)
+  __ := h
+
+@[simp]
+theorem isNthDegreeClosed_zero_iff_isRing {x : Nimber} : IsNthDegreeClosed 0 x ↔ IsRing x := by
+  refine ⟨IsNthDegreeClosed.toIsRing, fun h ↦ ⟨h, fun p ↦ ?_⟩⟩
+  cases _ : p.degree <;> aesop
+
+theorem IsNthDegreeClosed.toIsField {n : ℕ} {x : Nimber} (h : IsNthDegreeClosed n x) (hn : 1 ≤ n) :
+    IsField x := by
+  obtain hx₁ | hx₁ := le_or_gt x 1
+  · exact IsField.of_le_one hx₁
+  · refine ⟨h.toIsRing, fun y hy₀ hy ↦ ?_⟩
```

Comment:
Unroll the implicit `cases` instead of shrinking it with `obtain`

```suggestion
  cases le_or_gt x 1 with
  | inl hx₁ => exact IsField.of_le_one hx₁
  | inr hx₁ =>
    refine ⟨h.toIsRing, fun y hy₀ hy ↦ ?_⟩
```

Reply:
This "case on whether x ≤ 1 or 1 < x" thing is something I have to do a lot of times in subsequent PRs, and in all cases the x ≤ 1 branch is trivial or much easier, so I'd rather leave the compressed form.